### PR TITLE
Crossterm Refactors and 0.13 update

### PR DESCRIPTION
**Pull request by TimonPost**
- removed flush calls because `execute` already calls flush under the hood.
- moved some static functions into `From` traids
- removed useless clone in demo
- upgrade to crossterm 0.12
- upgrade to crossterm 0.13
- map all errors

### CODE REVIEW

**Comment by TimonPost** [comment]
Tomorrow, crossterm 0.13 is going to be released. Let's wait with this PR until then.

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,7 @@ travis-ci = { repository = "fdehau/tui-rs" }
 appveyor = { repository = "fdehau/tui-rs" }
 
 [features]
-default = ["termion"]
```
Comment (TimonPost) [inline Cargo.toml]: Is going to be changed back

File: Cargo.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ unicode-segmentation = "1.2"
 unicode-width = "0.1"
 termion = { version = "1.5" , optional = true }
 rustbox = { version = "0.11", optional = true }
-crossterm = { version = "^0.10", optional = true }
+crossterm = { version = "0.13.0", optional = true }
```
Comment (TimonPost) [inline Cargo.toml]:
```suggestion
crossterm = { version = "0.13", optional = true }
```

**Comment by heyrict** [comment]
Great PR :+1: ! I'd pretty like to see crossterm updated in tui, but it seems that the ci fails on errors fixed in the latest master. Can you rebase the PR to master and re-run the ci?

**Comment by TimonPost** [comment]
Let's wait another view days and we can integrate 0.14, it contains an awesome input rewrite and terminal resize events.

**Comment by TimonPost** [comment]
Although, feel free to merge it right now, because 0.14 is going to introduce more breaking changes. 
  @heyrict, @fdehau,  would be nice if a version with 0.13 can be released. A lot of users are waiting on this.

**Pull request merged**